### PR TITLE
Add local storage for user menu option and fix some small bugs

### PR DESCRIPTION
**Pull request by prestonfff**
- Add local storage for user menu option
- Fix some small issues on toggle and switching tabs

### CODE REVIEW

File: ArgusWeb/app/js/directives/charts/lineChart.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
 'use strict';
 
 angular.module('argus.directives.charts.lineChart', [])
-.directive('lineChart', ['$timeout', function($timeout) {
+.directive('lineChart', ['$timeout', 'Storage', '$routeParams', function($timeout, Storage, $routeParams) {
```
Comment (xizi-xu) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: There was an inputTracker service you could use if you want.

File: ArgusWeb/app/js/directives/charts/lineChart.js [hunk 2]
```diff
@@ -61,23 +63,38 @@ angular.module('argus.directives.charts.lineChart', [])
...
+                isTooltipOn : true
+            };
+
+            scope.dashboardId = $routeParams.dashboardId;
+            
+            var menuOption = Storage.get('menuOption_' + scope.dashboardId +'_' + lineChartIdName + scope.lineChartId);
+            if(menuOption){
+                    scope.menuOption = menuOption;
+            }
+
```
Comment (xizi-xu) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: I recommend line 80-93 to be moved to the controller of the directive. Here are the reasons:
1. The controller is called before the link and this ensures these variables got initiated before the DOM is populated. 
2. Variables in the controller can be used by other used by other directives, so things will be more flexible.
3. Scope manipulation should be done by the controller.

Comment (cnardi-dev) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: Good points here.  We can move forward with these changes in the next PR.

File: ArgusWeb/app/js/directives/charts/lineChart.js (hunk 1 above)
Comment (prestonfff) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: @xizi-xu  I see that. Just for this simple case, I don't need the default value for the get and set, so I just use the Storage and save some function call.

**Pull request merged**